pkgs/os-specific/darwin/apple-source-releases/adv_cmds/package.nix

```diff
@@ -0,0 +1,121 @@
+{
+  lib,
+  apple-sdk,
+  apple-sdk_11,
+  bison,
+  fetchFromGitHub,
+  flex,
+  mkAppleDerivation,
+  ncurses,
+  perl,
+  pkg-config,
+  stdenvNoCC,
+}:
+
+let
+  Libc = apple-sdk_11.sourceRelease "Libc";
+
+  # The headers have moved around between releases. Using one version is easier.
+  libplatform = fetchFromGitHub {
+    owner = "apple-oss-distributions";
+    repo = "libplatform";
+    rev = "libplatform-316.100.10";
+    hash = "sha256-gojt3sWOr7XO2yYI/B1CmNLTPFieSfoNtlOgQahOCok=";
+  };
+
+  # Use the same vendored version of xnu as system_cmds, so it only has to download once.
+  xnu = fetchFromGitHub {
+    owner = "apple-oss-distributions";
+    repo = "xnu";
+    rev = "xnu-8019.41.5";
+    hash = "sha256-CqKTYu10mTYXbEeZwDoDMw3+jKVLs/KSiAHU7nb0Lks=";
+  };
+
+  privateHeaders = stdenvNoCC.mkDerivation {
+    name = "adv_cmds-deps-private-headers";
+
+    buildCommand = ''
+      install -D -m644 -t "$out/include" \
+        '${libplatform}/private/_simple.h' \
+        '${Libc}/nls/FreeBSD/msgcat.h'
+
+      install -D -m644 -t "$out/include/System/sys" \
+        '${xnu}/bsd/sys/persona.h' \
+        '${xnu}/bsd/sys/proc.h'
+    '';
+  };
+in
+mkAppleDerivation {
+  releaseName = "adv_cmds";
+
+  outputs = [
+    "out"
+    "locale"
+    "ps"
+    "man"
+  ];
+
+  xcodeHash = "sha256-cpoHF++eko3NHGJlKnFONKnGkVRD0zI+bg/XLzWtpN8=";
+
+  postPatch = ''
+    # Meson generators require using @BASENAME@ in the output.
+    substituteInPlace mklocale/lex.l \
+      --replace-fail y.tab.h yacc.tab.h
+    substituteInPlace genwrap/lex.l \
+      --replace-fail y.tab.h genwrap.tab.h
+    substituteInPlace colldef/scan.l \
+      --replace-fail y.tab.h parse.tab.h
+
+    # Fix paths to point to the store
+    for file in genwrap.c genwrap.8; do
+      substituteInPlace genwrap/$file \
+        --replace-fail '/usr/local' "$out"
+    done
+
+    substituteInPlace localedef/localedef.pl \
+      --replace-fail '/usr/bin/perl' '${lib.getExe perl}' \
+      --replace-fail '/usr/bin' "$out/bin" \
+      --replace-fail '/usr/share/locale' "$locale/share/locale"
+  '';
+
+  env.NIX_CFLAGS_COMPILE = "-I${privateHeaders}/include";
+
+  buildInputs = [
+    # Use the 11.3 SDK because CMake depends on adv_cmds.ps, so it can’t simply be omitted when using an older SDK.
+    apple-sdk_11
+    ncurses
+  ];
+
+  nativeBuildInputs = [
+    bison
+    flex
+    perl
+    pkg-config
+  ];
+
+  mesonFlags = [
+    # Even though adv_cmds is built with a newer SDK, the default SDK is still the deployment target.
+    # Don’t built packages that use newer APIs unnecessarily.
+    (lib.mesonOption "sdk_version" (lib.getVersion apple-sdk))
```

Comment:
It’s the SDK used to build. I don’t do any back deployment stuff in this or system_cmds.

Reply:
Some of the commands require a newer SDK. I pass the default SDK version to avoid accidentally building them because they wouldn’t do any runtime availability checks because I didn’t backport them. I only did `ps` because it’s required by CMake.